src/pocapi.cpp

```diff
@@ -49,6 +49,34 @@ using ovms::StatusCode;
 using ovms::Timer;
 using std::chrono::microseconds;
 
+void OVMS_StatusDelete(OVMS_Status* status) {
+    if (status == nullptr)
+        return;
+    delete reinterpret_cast<ovms::Status*>(status);
+}
+
+OVMS_Status* OVMS_StatusGetCode(OVMS_Status* status,
+    uint32_t* code) {
+    if (status == nullptr)
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_STATUS));
+    if (code == nullptr)
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_STATUS_CODE_PTR));
+    ovms::Status* sts = reinterpret_cast<ovms::Status*>(status);
+    *code = static_cast<uint32_t>(sts->getCode());
+    return nullptr;
+}
+
+OVMS_Status* OVMS_StatusGetDetails(OVMS_Status* status,
+    const char** details) {
+    if (status == nullptr)
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_STATUS));
+    if (details == nullptr)
+        return reinterpret_cast<OVMS_Status*>(new Status(StatusCode::NONEXISTENT_STATUS_DETAILS_PTR));
+    ovms::Status* sts = reinterpret_cast<ovms::Status*>(status);
+    *details = sts->string().c_str();
```

Comment:
This line is correct as long as passed status is not ok - and it is probably issue on our side if someone here does have status ok - we don't return such statuses from C-API. However it could be good to add here additional safety check -> if status is ok then return INTERNAL_ERROR.

This is because status==OK does not have underlying message ptr so this could segfault in that case.

Reply:
There will never be segfault in case of OK status, the OK status has a string and it is empty string:
https://github.com/openvinotoolkit/model_server/blob/develop/src/status.cpp#L22

Added test for that.

Reply:
Check this line:
https://github.com/openvinotoolkit/model_server/blob/develop/src/status.hpp#L296